playwright/_impl/_async_base.py

```diff
@@ -56,10 +57,16 @@ def __str__(self) -> str:
         return self._impl_obj.__str__()
 
     def _async(self, api_name: str, coro: Awaitable) -> Any:
-        task = asyncio.current_task()
-        setattr(task, "__pw_api_name__", api_name)
-        setattr(task, "__pw_stack_trace__", traceback.extract_stack())
-        return coro
+        try:
+            logger.debug(f"=> {api_name} started")
+            task = asyncio.current_task()
+            setattr(task, "__pw_api_name__", api_name)
+            setattr(task, "__pw_stack_trace__", traceback.extract_stack())
+            logger.debug(f"<= {api_name} succeeded")
```

Comment:
Since you don't await coro in the middle, your log will be always saying started/succeeded w/o anything in between. The actual task will only start after you say succeeded.

Reply:
The coroutine is awaited in the generated in the `_generated.py` file hence that's why in the previous commit I had done logging in the docs generator. Will revert back to it since it works without any issues!

Reply:
You can also await it inside to resolve the Awaitable of the specific function inside it. Would look like that:

sync:
```py
            ...
            asyncio._set_running_loop(self._loop)
            return_value = task.result()
            logger.debug(f"<= {api_name} succeeded")
            return return_value
        except Exception as e:
            logger.debug(f"<= {api_name} failed")
            raise e
```

async:
```py
    async def _async(self, api_name: str, coro: Awaitable) -> Any:
        try:
            logger.debug(f"=> {api_name} started")
            task = asyncio.current_task()
            setattr(task, "__pw_api_name__", api_name)
            setattr(task, "__pw_stack_trace__", traceback.extract_stack())
            return_value = await coro
            logger.debug(f"<= {api_name} succeeded")
            return return_value
        except Exception as e:
            logger.debug(f"<= {api_name} failed")
            raise e
```

and add please a failed test with something like that:

```py
                   try:
                        page.wait_for_selector("foo", timeout=100)
                    except Error:
                        pass
```

which will result in a failed API call which throws and logs an failed log entry.